tests/rptest/scale_tests/tiered_storage_reader_stress_test.py

```diff
@@ -293,25 +301,39 @@ def stats_watcher():
             # all have been trimmed.
             assert stats['segment_readers']
 
-        def connections_closed():
+        def wait_for_stat(fn, label):
             for node, stats in self._get_stats().items():
-                if stats['connections'] > 0:
+                if metric := fn(stats):
                     self.logger.debug(
-                        f"Node {node.name} still has {stats['connections']} connections open"
-                    )
+                        f"Node {node.name} still has {metric} {label}")
                     return False
-
             return True
```

Comment:
nit: the wait_for_stat name clashes a bit with the purpose of the function.

reading the name i would expect that fn(stats) being true would make wait_for_stat true, but the logic is reversed

this stumped me a bit when reading 

```python 
        # No new downloads should be started once all reader activity is done
        for _ in range(10):
            time.sleep(0.5)
            assert wait_for_stat(
                lambda s: s['hydrations_count'], 'active hydrations'
            ), 'found an active hydration where none expected'
```

Reply:
good point, renamed to `is_metric_zero`